### PR TITLE
Pass --cap-lints=allow instead of -Awarnings

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit adds support to Cargo to pass `--cap-lints allow` to all upstream
dependencies instead of `-A warings`. This should serve the same purpose of
suppressing warnings in upstream dependencies as well as preventing widespread
breakage whenever a change to a lint is introduced in the compiler.

### CODE REVIEW

**Comment by alexcrichton** [comment]
r? @brson 

cc https://github.com/rust-lang/rust/pull/27260

File: src/cargo/ops/cargo_rustc/mod.rs [hunk 1]
```diff
@@ -336,10 +313,14 @@ fn rustc(package: &Package, target: &Target, profile: &Profile,
...
-                            is_path_source;
-        if !show_warnings {
-            rustc.arg("-Awarnings");
+        let allow_warnings = package.package_id() == cx.resolve.root() ||
+                             is_path_source;
+        if !allow_warnings {
+            if cx.config.rustc_info().cap_lints {
+                rustc.arg("--cap-lints").arg("allow");
+            } else {
+                rustc.arg("-Awarnings");
```
Comment (huonw) [inline src/cargo/ops/cargo_rustc/mod.rs]: It's slightly amusing that the `--allow warnings` flag is being passed when `!allow_warnings`.

**Comment by alexcrichton** [comment]
ping r? @brson

**Pull request merged**